mllib/src/main/scala/org/apache/spark/ml/r/RWrapperUtils.scala

```diff
@@ -35,13 +35,37 @@ object RWrapperUtils extends Logging {
    */
   def checkDataColumns(rFormula: RFormula, data: Dataset[_]): Unit = {
     if (data.schema.fieldNames.contains(rFormula.getLabelCol)) {
-      logWarning("data containing 'label' column, so change its name to avoid conflict")
-      rFormula.setLabelCol(rFormula.getLabelCol + "_output")
+      val newLabelName = convertToUniqueName(rFormula.getLabelCol, data.schema.fieldNames)
```

Comment:
I think in `if (data.schema.fieldNames.contains(rFormula.getFeaturesCol))`, it's checking `label` only

and in `convertToUniqueName ()`,  `_output` will be appended resulting in `label_output `: `var newName = originalName + "_output"`, and then `label_output ` is checked at `while (fieldNames.contains(newName))`